### PR TITLE
fix new-nugetpackage to cope with more output from child process

### LINKED ISSUE #478: Can't publish module to local repository since v2.1.3
Steps to reproduce
------------------
Register local repository:

```powershell
$repo = @{
Name='Test'
SourceLocation='D:\Test'
PublishLocation='D:\Test'
InstallationPolicy='trusted'
}
Register-PSRepository @repo
```
Try to publish any module to local repository:
```powershell
Publish-Module -Repository Test -Verbose -Name Pester
```

Expected behavior
-----------------
Module is published successfully on PowerShellGet v2.1.2
powershellget_2_1_2

Actual behavior
---------------
Publish-Module hangs indefinitely on PowerShellGet v2.1.3
powershellget_2_1_3

Environment data
----------------

```powershell
> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      5.1.14393.2879
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.14393.2879
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

```powershell

> Get-Module -ListAvailable PowerShellGet,PackageManagement

    Directory: C:\Program Files\WindowsPowerShell\Modules

ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.3.2      PackageManagement                   {Find-Package, Get-Package, Get-PackageProvider, Get-PackageSource...}
Script     2.1.2      PowerShellGet                       {Find-Command, Find-DSCResource, Find-Module, Find-RoleCapability...}
```

```powershell
> Get-PackageProvider -ListAvailable

Name                     Version          DynamicOptions
----                     -------          --------------
msi                      3.0.0.0          AdditionalArguments
msu                      3.0.0.0
nuget                    2.8.5.208
NuGet                    3.0.0.1          Destination, ExcludeVersion, Scope, SkipDependencies, Headers, FilterOnTag, Contains, AllowPrereleaseVer... PowerShellGet            2.1.2.0          PackageManagementProvider, Type, Scope, AllowClobber, SkipPublisherCheck, InstallUpdate, NoPathUpdate, A...
Programs                 3.0.0.0          IncludeWindowsInstaller, IncludeSystemComponent

```

### PR DESCRIPTION
**Pull request by edyoung**
Should resolve #478.

### CODE REVIEW

**Comment by edyoung** [comment]
FYI @Benny1007

File: src/PowerShellGet/private/functions/New-NugetPackage.ps1 [hunk 1]
```diff
@@ -27,89 +29,83 @@ function New-NugetPackage {
...
-        $process = New-Object System.Diagnostics.Process
-        $process.StartInfo = $processStartInfo
-        $process.Start() | Out-Null
-        $process.WaitForExit()
+    # get any remaining output
+    $process.WaitForExit()
+    $outputLines += $process.StandardOutput.ReadToEnd()
 
-        if (Test-Path -Path $tempPath) {
-            Remove-Item -Path $tempPath -Force -Recurse
```
Comment (alerickson) [inline src/PowerShellGet/private/functions/New-NugetPackage.ps1:97]: This should still be left in-- we should still clean up the temp path.

**Review by alerickson — APPROVED** [review]

File: src/PowerShellGet/private/functions/New-NugetPackage.ps1 [hunk 2]
```diff
@@ -27,89 +29,83 @@ function New-NugetPackage {
...
+    $stdOut = $outputLines -join "`n"
 
-        if (-Not ($process.ExitCode -eq 0 )) {
-            $stdOut = $process.StandardOutput.ReadToEnd()
-            throw "dotnet cli failed to pack $stdOut"
-        }
+    Write-Verbose "finished running $($processStartInfo.FileName) with exit code $($process.ExitCode)"
 
+    if (-Not ($process.ExitCode -eq 0 )) {
+        $stdErr = $process.StandardError.ReadToEnd()
```
Comment (Benny1007) [inline src/PowerShellGet/private/functions/New-NugetPackage.ps1]: My experience with the dotnet core cli was that it didn't write it's errors to the StandardError stream which was the reason for $process.StandardOutput.ReadToEnd() before. Has this been fixed?

Comment (edyoung) [reply src/PowerShellGet/private/functions/New-NugetPackage.ps1]: I don't believe so, switched it

**Pull request merged**